ledger/sandbox/src/main/resources/db/migration/V5__Add_packages.sql

```diff
@@ -0,0 +1,31 @@
+-- Copyright (c) 2019 Digital Asset (Switzerland) GmbH and/or its affiliates. All rights reserved.
+-- SPDX-License-Identifier: Apache-2.0
+
+---------------------------------------------------------------------------------------------------
+-- V5: List of packages
+--
+-- This schema version adds a table for tracking DAML-LF packages.
+-- Previously, packages were only stored in memory and needed to be specified through the CLI.
+---------------------------------------------------------------------------------------------------
+
+
+
+CREATE TABLE packages (
+  -- The unique identifier of the package (the hash of its content)
+  package_id         varchar primary key   not null,
+  -- Packages are uploaded as DAR files (i.e., in groups)
+  -- This field can be used to find out which packages were uploaded together
+  submission_id      varchar               not null,
```

Comment:
I think we ought to make `submission_id` nullable, as the API Server will not have a submission identifier available through the state updates.

Reply:
This ID is to be able to tell what packages were uploaded together, in case of two concurrent `uploadPackages` calls (those would share the same `ledger_offset`, and, if extremely unlucky, the same `known_since` value). It is a concept of the DAO, and has nothing to do with participant state or ledger API interfaces. A random UUID is generated at each call site of `LedgerDao.uploadLfPackages`.

Reply:
Okay, thank you for the explanation. I think the name is slightly misleading, as it's not really a submission identifier, but rather a receipt or bundle identifier. My initial thought was that this refers to a user-provided identifier upon package upload via the ledger api.
Other than maybe `local_submission_id` I don't really have a better name.

Reply:
`upload_id`?

Reply:
Thank you. I'm gonna go and facepalm about myself for a while ;)